e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -92,6 +92,13 @@ def __init__(
         if self.parameter.sets[0] in ["diurnal_cycle", "arm_diags"]:
             self.is_sub_monthly = True
 
+        # A boolean to keep track of whether the source variable(s) for a
+        # target variable contains a wildcard ("?"). This is used to determine
+        # whether to pass a tuple of wildcard variables as args to derived
+        # variable function (True), or to unpack an expected number of variables
+        # as args to a derived variable function (False).
+        self.is_src_vars_wildcard = False
```

Comment:
`is_srcs_vars_wildcard` is a new attribute that is needed to determine whether to pass a list of variables to the derived variable function (`True`), or to unpack the list of variables as individual args (`False`, default).

If the source variables are derived using a wildcard (`"?"`), then there can be N number of variables which we can't pre-define for a derived variable function. Therefore, we need to pass the entire list of variables for these cases.